pkg/dao/schema/resource_definition_matching_rule.go

```diff
@@ -0,0 +1,87 @@
+package schema
+
+import (
+	"entgo.io/ent"
+	"entgo.io/ent/dialect/entsql"
+	"entgo.io/ent/schema"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+	"entgo.io/ent/schema/index"
+	
+	"github.com/seal-io/walrus/pkg/dao/entx"
+	"github.com/seal-io/walrus/pkg/dao/schema/mixin"
+	"github.com/seal-io/walrus/pkg/dao/types"
+	"github.com/seal-io/walrus/pkg/dao/types/object"
+	"github.com/seal-io/walrus/pkg/dao/types/property"
+)
+
+type ResourceDefinitionMatchingRule struct {
+	ent.Schema
+}
+
+func (ResourceDefinitionMatchingRule) Mixin() []ent.Mixin {
+	return []ent.Mixin{
+		mixin.ID(),
+		mixin.Time().WithoutUpdateTime(),
+	}
+}
+
+func (ResourceDefinitionMatchingRule) Indexes() []ent.Index {
+	return []ent.Index{
+		index.Fields("resource_definition_id", "template_id", "name").
+			Unique(),
+	}
+}
+
+func (ResourceDefinitionMatchingRule) Fields() []ent.Field {
+	return []ent.Field{
+		object.IDField("resource_definition_id").
+			Comment("ID of the resource definition to which the relationship connects.").
+			NotEmpty().
+			Immutable(),
+		object.IDField("template_id").
+			Comment("ID of the template version to which the relationship connects.").
+			NotEmpty().
+			Immutable(),
+		field.String("name").
+			Comment("Name of the matching rule.").
+			NotEmpty().
+			Immutable().
+			Annotations(
+				entx.Input(entx.WithUpdate())),
+		field.JSON("selector", types.Selector{}).
+			Comment("Resource selector."),
+		property.ValuesField("attributes").
+			Comment("Attributes to configure the template.").
+			Optional(),
+	}
+}
+
+func (ResourceDefinitionMatchingRule) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.To("resource_definition", ResourceDefinition.Type).
+			Field("resource_definition_id").
+			Comment("Resource definition that connect to the relationship.").
+			Unique().
+			Required().
+			Immutable().
+			Annotations(
+				entsql.OnDelete(entsql.Cascade),
+				entx.SkipIO()),
+		edge.To("template", TemplateVersion.Type).
+			Field("template_id").
```

Comment:
template_version_id?

Reply:
To align with https://github.com/seal-io/walrus/blob/main/pkg/dao/schema/service.go#L74

Reply:
Do you think we should update the matching rule and resource to use template_version_id, since it's the template version ID actually.

Reply:
Not in this PR